services/infra/InfraService.ts

```diff
@@ -11,41 +12,87 @@ interface InfraServiceProps {
   sitesService: SitesService
   reposService: ReposService
   deploymentsService: DeploymentsService
+  mailer: MailClient
 }
 
 export default class InfraService {
-  private readonly usersService: InfraServiceProps["usersService"]
+  private readonly usersService
 
-  private readonly sitesService: InfraServiceProps["sitesService"]
+  private readonly sitesService
 
-  private readonly reposService: InfraServiceProps["reposService"]
+  private readonly reposService
 
-  private readonly deploymentsService: InfraServiceProps["deploymentsService"]
+  private readonly deploymentsService
+
+  private readonly mailer
 
   constructor({
     usersService,
     sitesService,
     reposService,
     deploymentsService,
+    mailer,
   }: InfraServiceProps) {
     this.usersService = usersService
     this.sitesService = sitesService
     this.reposService = reposService
     this.deploymentsService = deploymentsService
+    this.mailer = mailer
+  }
+
+  sendCreateError = async (
+    email: string,
+    repoName: string | undefined,
+    submissionId: string | undefined,
+    error: string
+  ) => {
+    const subject = `[Isomer] Create site ${repoName} FAILURE`
+    const html = `<p>Isomer site ${repoName} was <b>not</b> created successfully. (Form submission id [${submissionId}])</p>
+<p>${error}</p>
+<p>This email was sent from the Isomer CMS backend.</p>`
+    await this.mailer.sendMail(email, subject, html)
+  }
+
+  sendCreateSuccess = async (
+    email: string,
+    repoName: string,
+    submissionId: string | undefined,
+    stagingUrl: string,
+    productionUrl: string
+  ) => {
+    const subject = `[Isomer] Create site ${repoName} SUCCESS`
+    const html = `<p>Isomer site ${repoName} was created successfully. (Form submission id [${submissionId}])</p>
+<p>You may now view this repository on GitHub. <a href="${stagingUrl}">Staging</a> and <a href="${productionUrl}">production</a> deployments should be accessible within a few minutes.</p>
+<p>This email was sent from the Isomer CMS backend.</p>`
+    await this.mailer.sendMail(email, subject, html)
   }
 
   createSite = async ({
-    email,
+    requesterEmail,
     repoName,
+    submissionId,
   }: {
-    email: string
-    repoName: string
+    requesterEmail: string
+    repoName: string | undefined
+    submissionId: string | undefined
   }) => {
```

Comment:
the precondition to this working seems to be that we provide a `repoName` and `submissionId` - we could validate this prior to invoking this method so that this method can focus solely on business logic.

also, `submissionId` seems to be used solely for tracking/debugging purposes - perhaps we could defer the responsibility of sending mail up a level and keep this simple?

Reply:
I'll use a standard argument list instead of an object.

I'll move other FormSG related stuff from InfraService into the router, which is FormSG specific. That should take care of your second comment.